x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/InferenceStats.java

```diff
@@ -226,24 +234,67 @@ public Accumulator merge(InferenceStats otherStats) {
         }
 
         public Accumulator incMissingFields() {
-            this.missingFieldsAccumulator.increment();
-            return this;
+            readWriteLock.readLock().lock();
+            try {
+                this.missingFieldsAccumulator.increment();
+                return this;
+            } finally {
+                readWriteLock.readLock().unlock();
+            }
         }
 
         public Accumulator incInference() {
-            this.inferenceAccumulator.increment();
-            return this;
+            readWriteLock.readLock().lock();
+            try {
+                this.inferenceAccumulator.increment();
+                return this;
+            } finally {
+                readWriteLock.readLock().unlock();
+            }
         }
 
         public Accumulator incFailure() {
-            this.failureCountAccumulator.increment();
-            return this;
+            try {
+                readWriteLock.readLock().lock();
+                this.failureCountAccumulator.increment();
+                return this;
+            } finally {
+                readWriteLock.readLock().unlock();
+            }
         }
 
-        public InferenceStats currentStats() {
+        private InferenceStats currentStats() {
```

Comment:
There is only one place where this method is called so I think it could be inlined there.